test/e2e/system/system_utils.go

```diff
@@ -20,9 +20,9 @@ import (
 	"strings"
 )
 
-// IsMasterNode returns true if given node is a registered master.
-// TODO: find a better way of figuring out if given node is a registered master.
-func IsMasterNode(nodeName string) bool {
+// DeprecatedIsMasterNode returns true if given node is a registered master.
+// DEPRECATED: use a label selector provided by test initialization.
```

Comment:
As this is now in the tests (yay!), and in the absence of a proposal on how to deal with e2e tests that act on the control plane, it feels wrong to deprecate this.

Propose we move the label to a flag, and put a warning like "prefer other mechanisms to atomically considering a node to be master or not".

Reply:
Why is it wrong to deprecate?  Something that is broken is broken - this check doesn't even work on a subset of conformant clusters, so sig-testing definitely needs to help identify next steps.  This is being explicitly marked this way to clearly indicate that no one should add new code using this, which is something Jordan and I have seen repeatedly over the last month.

Reply:
The proposal on how to act on e2e tests on the control plane is to accept labels that identify masters as a parameter (defaulting to existing), if no masters are present the test should skip or continue (depending), and the use of that flag is considered what this is deprecated in favor of.